expr/functions/nPercentile/function.go

```diff
@@ -44,10 +45,10 @@ func (f *nPercentile) Do(ctx context.Context, e parser.Expr, from, until int64,
 
 	results := make([]*types.MetricData, len(arg))
 	for i, a := range arg {
-		r := *a
-		r.Name = "nPercentile(" + a.Name + "," + percentStr + ")"
+		r := a.CopyLink()
+		r.Name = fmt.Sprintf("nPercentile(%s,%s)", a.Name, percentStr)
 		r.Values = make([]float64, len(a.Values))
-
+		r.Tags["nPercentile"] = fmt.Sprintf("%f", percent)
```

Comment:
May be reuse already buffered to string conversion
r.Tags["nPercentile"] = percentStr